Tests/SwiftSyntaxBuilderTest/StringLiteralTests.swift

```diff
@@ -170,4 +170,182 @@ final class StringLiteralTests: XCTestCase {
+    let buildable = StringLiteralSegmentsSyntax {
+      StringSegmentSyntax(content: .stringSegment(#"Error validating child at index \(index) of \(nodeKind):"#), trailingTrivia: .newline)
+      StringSegmentSyntax(content: .stringSegment(#"Node did not satisfy any node choice requirement."#), trailingTrivia: .newline)
+      StringSegmentSyntax(content: .stringSegment(#"Validation failures:"#), trailingTrivia: .newline)
+      StringSegmentSyntax(content: .stringSegment(#"\(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))"#))
+    }
+
+    AssertBuildResult(
+      buildable,
+      #"""
+      Error validating child at index \(index) of \(nodeKind):
+      Node did not satisfy any node choice requirement.
+      Validation failures:
+      \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))
+      """#
+    )
+  }
+
+  func testMultiLineStringWithResultBuilder() {
+    let buildable = StringLiteralExprSyntax(
+      openQuote: .multilineStringQuoteToken(trailingTrivia: .newline),
+      segments: StringLiteralSegmentsSyntax {
+        StringSegmentSyntax(content: .stringSegment(#"Error validating child at index \(index) of \(nodeKind):"#), trailingTrivia: .newline)
+        StringSegmentSyntax(content: .stringSegment(#"Node did not satisfy any node choice requirement."#), trailingTrivia: .newline)
+        StringSegmentSyntax(content: .stringSegment(#"Validation failures:"#), trailingTrivia: .newline)
+        StringSegmentSyntax(content: .stringSegment(#"\(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))"#))
+      },
+      closeQuote: .multilineStringQuoteToken(leadingTrivia: .newline)
+    )
+
+    AssertBuildResult(
+      buildable,
+      #"""
+      """
+      Error validating child at index \(index) of \(nodeKind):
+      Node did not satisfy any node choice requirement.
+      Validation failures:
+      \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))
+      """
+      """#
+    )
+  }
+
+  func testMultiLineStringWithFunctions() {
+    let buildable = StringLiteralExprSyntax(
+      openQuote: .multilineStringQuoteToken(trailingTrivia: .newline),
+      content:
+        #"""
+        assertionFailure("""
+          Error validating child at index \(index) of \(nodeKind):
+          Node did not satisfy any node choice requirement.
+          Validation failures:
+          \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n") )
+          """, file: file, line: line)
+        """#
+      ,
+      closeQuote: .multilineStringQuoteToken(leadingTrivia: .newline)
+    )
+
+    AssertBuildResult(
+      buildable,
+      ##"""
+      #"""
+      assertionFailure("""\#n  Error validating child at index \(index) of \(nodeKind):\#n  Node did not satisfy any node choice requirement.\#n  Validation failures:\#n  \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n") )\#n  """, file: file, line: line)
```

Comment:
Why do you use escaped `\#n` instead of just inserting a newline into the string literal?

Reply:
It was because the outcome is that. 

```
testMultiLineStringWithFunctions(): failed - Actual output (+) differed from expected output (-):
–assertionFailure("""
–Error validating child at index \(index) of \(nodeKind):
–Node did not satisfy any node choice requirement.
–Validation failures:
–  \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n") )
–""", file: file, line: line)
+#"""
+assertionFailure("""\#n  Error validating child at index \(index) of \(nodeKind):\#n  Node did not satisfy any node choice requirement.\#n  Validation failures:\#n  \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n") )\#n  """, file: file, line: line)
+"""#
```

Not sure why?

Reply:
Oh, interesting. So the actual output contains `\#n`. That shouldn’t be the case.

But that might be a side effect of you creating a single string segment that contains multiple lines and a string interpolation segment. The issue might just disappear if you construct the `StringLiteralExprSyntax` by parsing a string.

Reply:
Will remove the test as I understand and the usage of the API it looks like wrong usage. 
And I think tests should show correct examples